content/blog/text-rendering-updates/index.Rmd

```diff
@@ -0,0 +1,171 @@
+
+Again, it is unlikely that you will come into contact with textshaping directly so you will mostly experience these improvements in the way text just appears more correct (to the extend that this was ever an issue for you). The place you are most likely to stumble upon these changes is marquee, which uses textshaping under the hood. Styling in marquee has been expanded to include a `text_direction` setting. It defaults to `"auto"` which mean "deduce it from the text you get", but you can also set it to `"ltr"` or `"rtl"` to set the direction explicitly. Be aware that this setting doesn't change how single glyphs flow so you cannot use it to e.g. write arabic in left-to-right flow. Instead it governs the paragraph-level direction and thus how bi-directional text should be assembled. It also governs to which side indentation happen and the placement of bullets in bullet lists. Often, leaving it on the default value will work fine. There are also two new values for the `align` setting. `"auto"` picks either `"left"` or `"right"` depending on the text direction, while `"justified"` picks either `"justified-left"` or `"justified-right"`. This makes it much easier to work natively with right-to-left text as everything just looks as it should. To top it off, `classic_style()` gains an `ltr` argument that controls whether the styling in general should cater to left-to-right or right-to-left text. It controls things such as the position of the grey bar in quotation blocks and the indentation of nested lists.
+
+## A marquee for Everyone
+
+Speaking of marquee, the biggest obstacle it has put in front of its users is that it is build on very new features in R. The ability to write text by placing glyphs one at a time was only added in R 4.2 and not every graphics device supports it yet (worse still, the implementation in the default Mac quartz device caused the session to crash). Again, ragg is your friend, but the Cairo devices also has excellent support.
+
+Text rendering, however, should always work. It is quite frustrating for text to not show up when you expect it to. Because of this it has been a clear plan to expand the support for marquee somehow. With the new version of marquee this is finally a reality. How does it work? Well, remember when we talked about extracting glyph outlines and rasters? If marquee encounters a graphics device that doesn't provide the necessary features it will take matter into it's own hand, extract all the necessary polygons and bitmaps and plot them. It is certainly not faster than relying on the optimized routines of the graphics device and it can also lead to visual degradation at smaller font sizes. But it works - everywhere.
```

Comment:
The latest version of marquee solves these compatibility issues by taking a clever approach: when it encounters a device that doesn't support modern text features, it automatically switches to rendering text using primitive shapes and images. Here's how it works: